roles/neutron-data-network/tasks/main.yml

```diff
@@ -70,17 +70,30 @@
     group: neutron
     mode: 0755
 
-- name: modified neutron lbaas templates to workaround 2000 connection limit
-  copy:
-    src: etc/neutron/lbaas_templates/{{ item }}
-    dest: /etc/neutron/lbaas_templates/{{ item }}
+- name: template neutron lbaas haproxy
+  template:
+    src: etc/neutron/haproxy_base.j2
+    dest: /etc/neutron/lbaas_templates/haproxy_base.j2
     owner: root
     group: neutron
     mode: 0644
+  notify: restart neutron lbaas agent
+  when: (neutron.lbaas.enabled == "smart" and
+         groups['controller'][0] not in groups['compute']) or
+         neutron.lbaas.enabled|bool
+
+- name: symlink other neutron lbaas haproxy templates
+  file:
+    src: /opt/openstack/current/neutron/lib/python2.7/site-packages/neutron_lbaas/services/loadbalancer/drivers/haproxy/templates/{{ item }}
```

Comment:
good catch, we don't have the code here to remove the non-link files.

As for discovering the path, I think it's find to be coded as is, because when the path changes we want to know about it and re-evaluate what we're doing.